linux/ibus-keyman/src/engine.c

```diff
@@ -27,6 +27,16 @@
 
 #include <gdk/gdk.h>
 #include <gio/gio.h>
+
+#ifdef GDK_WINDOWING_X11
+#include <gdk/gdkx.h>
+#include <X11/XKBlib.h>
+#endif
+
+#ifdef GDK_WINDOWING_WAYLAND
+#include <gdk/gdkwayland.h>
+#endif
```

Comment:
This seems to be mixing wayland support into this PR? Is it possible to leave out all new wayland-related code and put that into a separate PR?

Do we see scenarios where both wayland and x11 need to be supported simultaneously?

Reply:
> Do we see scenarios where both wayland and x11 need to be supported simultaneously?

yes, the way current Ubuntu versions are configured is that it they build with Wayland, but also have the X11 support enabled for apps that don't support Wayland yet. And I think this will continue for several more years...